Client/Frontend/Home/Wallpapers/v2/NetworkServices/WallpaperNetworkModule.swift

```diff
@@ -0,0 +1,39 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+class WallpaperNetworkingModule: WallpaperNetworking {
+
+    /// A basic async/await wrapper
+    func data(from url: URL) async throws -> (Data, URLResponse) {
+        return try await withCheckedThrowingContinuation { continuation in
+            URLSession.shared.dataTask(with: url) { data, response, error in
+
+                if let error = error {
+                    continuation.resume(throwing: error)
+                    return
+                }
+
+                guard let response = response as? HTTPURLResponse,
+                      (200...299).contains(response.statusCode)
+                else {
```

Comment:
Might be worth it to use an existing method we have to `validatedHTTPResponse`  which is defined in NetworkUtils.swift